js/templates/modals/listingDetail/listing.html

```diff
@@ -106,23 +106,34 @@ <h2 class="flexExpand js-cryptoTitle cryptoTitle"></h2>
               </div>
             <% }); %>
             <%
+              const priceCryptoCode = ob.currencyMod.getCryptoCurByCode(ob.price.currencyCode);
               let tip;
               let buyNowClass = 'disabled';
               let buyNowTranslationKey = ob.metadata.contractType !== 'CRYPTOCURRENCY' ?
                 'listingDetail.buyNow' :
                 'listingDetail.buyCryptoNow';
               let unpurchaseable = true;
-              const serverCur = ob.currencyMod.getServerCurrency();
-              const serverCurCode = serverCur.isTestnet ? serverCur.testnetCode : serverCur.code;
 
-              if (ob.metadata.acceptedCurrencies.indexOf(serverCurCode) === -1) {
+              if (!ob.crypto.anySupportedByWallet(ob.metadata.acceptedCurrencies)) {
                 tip = ob.polyT('listingDetail.unableToPurchase.incompatibleCrypto',
                   {
-                    acceptedCurrencies: ob.metadata.acceptedCurrencies.join(', '),
-                    serverCur: serverCurCode,
+                    acceptedCurs: ob.metadata.acceptedCurrencies.join(', '),
+                    walletCurs: ob.crypto.supportedWalletCurs()
+                      .join(', '),
                   });
               } else if (ob.metadata.contractType !== 'CRYPTOCURRENCY' &&
+                !(
+                  priceCryptoCode &&
+                  ob.crypto.anySupportedByWallet([priceCryptoCode.code, priceCryptoCode.testnetCode])
+                ) &&
                 ob.currencyMod.getCurrencyValidity(ob.price.currencyCode) !== 'VALID') {
+                // Ensure we have exchange rate data based on the currency the listing is listed
+                // in. Otherwise, it is not purchaseable since the price cannot be determined.
+                // If the listing price code is one of the accepted currencies AND the user pays in
+                // that currency, then exchange rate data is not needed.
+                // TODO:
+                // TODO:
+                // TODO: ugh - how to handle that case?
```

Comment:
This is a pretty interesting scenario, albeit quite the edge case, so we should probably take a lower scope path.

Anyhow, the idea is that if a listing is priced in a certain currency we don't have exchange rate data for then the Buy Now button is disabled. But, if the listing happens to be priced in one of the accepted currencies (and the accepted currency is supported by the user's wallet), then as long as the user pays in that currency, the purchase would work. So maybe in this case, we leave the button enabled:

![image](https://user-images.githubusercontent.com/794517/45785571-0f87b380-bc2a-11e8-9132-d462cfcc814f.png)

And then in the Checkout, maybe we just disable all options other than the one in the pricing currency? Maybe with a little 'exchange rate unavailable` text, similar to how `Not available` is used when your wallet doesn't support one of the accepted currencies?

![image](https://user-images.githubusercontent.com/794517/45785602-35ad5380-bc2a-11e8-8d08-ff4bbdbd59d0.png)

(just want to stress to not put too much effort into this because most listings are probably priced in fiat AND the exchange rate API shouldn't be down too often (hopefully))

Reply:
I think it would be nice to allow like to like crypto purchases (ie if the listing is priced in a crypto coin, I can buy it with that crypto coin), since that would allow the network to keep working if the ticker was down, but not if it adds a lot of complexity.

Your idea seems right to me, we'd just disable all the currencies that won't work with a note next to each explaining why.